src/error.rs

```diff
@@ -133,25 +134,66 @@ pub trait ErrorDomain: Copy {
 
 /// Generic error used for functions that fail without any further information
 #[derive(Debug)]
-pub struct BoolError(pub &'static str);
+pub struct BoolError(pub Cow<'static, str>);
```

Comment:
Maybe while we extend this, we could also add a field for `file!()`, `module_path!()` and `line!()` and make a constructor-macro that fills that in?

Reply:
@GuillaumeGomez @EPashkin opinions?

Reply:
Good idea!

Reply:
See the new commit. I'm not convinced by the macro name: `glib_result_from_glib`. Any idea? I though `glib_bool_error_result_from_glib` was a bit too long. Maybe `glib_result_from_glib_bool`?

Reply:
`glib_bool_error!(message)` and `glib_bool_error_from_gboolean!(message, b)` maybe?

Reply:
I would tend to use `glib_result_from_gboolean` instead of `glib_bool_error_from_gboolean` since the return value is a `Result`.

Why do you prefer putting the boolean as the last argument?

Reply:
I don't mind either way :)